File: via_l2g/最长上升子序列.py
def max_before(a, b, i):
    l = len(a)
    m = 0
    for j in range(l):
        if b[j] < i and a[j] > m:
            m = a[j]
    return m


class Solution(object):
    def lengthOfLIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        lis = [0] * len(nums)
        lis[0] = 1
        for i in range(1, len(nums)):
            lis[i] = max_before(lis[:i], nums[:i], nums[i]) + 1
        return max(lis)
        

File: via_l2g/最长上升子序列.cs
public class Solution {
    public int LengthOfLIS(int[] nums) {
        if (nums.Length == 0) return 0;
        int [] lim = new int[nums.Length];
        int m, cur;
        lim[0] = 1;
        cur = 1;
        for (int i = 1; i < nums.Length; i++) {
            m = 0;
            for (int j = 0; j < i; j++) {
                if (nums[j] < nums[i] && lim[j] > m) {
                    m = lim[j];
                }
            }
            lim[i] = m + 1;
            if (lim[i] > cur) cur = lim[i];
        }
        return cur;
    }
}
